### DISCUSSION TITLE
List all packages and show if update is available - Export list as JSON

### DISCUSSION THREAD

**Discussion by Ogglas** [renovatebot/renovate#16007] [Request Help]
Is it possible to use Renovate to simply list list all packages and any update available to them and export the result to a JSON file or similar? 

I'm thinking something like `npm-check-updates` and `dotnet list package --outdated` but for every dependency and also include packages not requiring an update.  

I can get a list but today I have to perform separate commands:

For npm:

`npm list`

![image](https://user-images.githubusercontent.com/4015237/173155319-a5189d4b-5209-428d-8977-44306ada05f4.png)

https://nodejs.dev/learn/find-the-installed-version-of-an-npm-package

`npm-check-updates`

![image](https://user-images.githubusercontent.com/4015237/173155388-e1403d12-c21f-4c11-a394-9302a3b55e5c.png)

https://www.npmjs.com/package/npm-check-updates

NuGet:

`dotnet list package `

`dotnet list package --outdated`

![image](https://user-images.githubusercontent.com/4015237/173156026-d1463e5d-d73e-48c6-a65f-84791d188550.png)

![image](https://user-images.githubusercontent.com/4015237/173155877-8a3b9de3-b369-4f09-9ded-bec05ce6dee2.png)

https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-list-package

I then have to convert the result to a JSON file in this case. I can of course also parse `.sln`, `.csproj`, `packages.lock.json`, `package.json` and` package-lock.json` files but if Renovate already handles this it would be great.

**Comment by rarkins** [comment]
If you run with `--dry-run=lookup` and then you something similar to what you want logged. I forget if it's INFO or DEBUG level though

**Reply by Ogglas** [reply]
> If you run with `--dry-run=lookup` and then you something similar to what you want logged. I forget if it's INFO or DEBUG level though

Thanks! Will try this out. From docs: _Information provided mainly in debug log level._

https://docs.renovatebot.com/self-hosted-configuration/#dryrun
